src/main/java/edu/harvard/iq/dataverse/harvest/client/HarvestingClientServiceBean.java

```diff
@@ -198,28 +190,40 @@ public void setHarvestFailure(Long hcId, Date currentTime) {
         ClientHarvestRun currentRun = harvestingClient.getLastRun();
         
         if (currentRun != null && currentRun.isInProgress()) {
-            // TODO: what if there's no current run in progress? should we just
-            // give up quietly, or should we make a noise of some kind? -- L.A. 4.4      
             
-            currentRun.setFailed();
-            currentRun.setFinishTime(currentTime);
+            currentRun.setResult(result);
+            currentRun.setFinishTime(finishTime);
+            currentRun.setHarvestedDatasetCount(Long.valueOf(harvestedCount));
+            currentRun.setFailedDatasetCount(Long.valueOf(failedCount));
+            currentRun.setDeletedDatasetCount(Long.valueOf(deletedCount));
         }
-    }  
+    }
+    
+    public Long getNumberOfHarvestedDatasetsByAllClients() {
+        try {
+            return (Long) em.createNativeQuery("SELECT count(d.id) FROM dataset d "
+                    + " WHERE d.harvestingclient_id IS NOT NULL").getSingleResult();
+
+        } catch (Exception ex) {
+            logger.info("Warning: exception looking up the total number of harvested datasets: " + ex.getMessage());
+            return 0L;
+        }
+    }
     
     public Long getNumberOfHarvestedDatasetByClients(List<HarvestingClient> clients) {
-        String dvs = null; 
+        String clientIds = null; 
         for (HarvestingClient client: clients) {
-            if (dvs == null) {
-                dvs = client.getDataverse().getId().toString();
+            if (clientIds == null) {
+                clientIds = client.getId().toString();
             } else {
-                dvs = dvs.concat(","+client.getDataverse().getId().toString());
+                clientIds = clientIds.concat(","+client.getId().toString());
             }
         }
         
         try {
-            return (Long) em.createNativeQuery("SELECT count(d.id) FROM dataset d, "
-                    + " dvobject o WHERE d.id = o.id AND o.owner_id in (" 
-                    + dvs + ")").getSingleResult();
+            return (Long) em.createNativeQuery("SELECT count(d.id) FROM dataset d "
+                    + " WHERE d.harvestingclient_id in (" 
```

Comment:
I can see where the join of dataset and dvobject isn't needed, but why the switch to harvestingclient_id? Was the old code just wrong or is this a typo?

Reply:
(Sorry for the delay)
The new query is simpler, and correct, all the harvested datasets are uniquely tied to the specific harvesting configurations (clients). 

The old query must have followed the logic from pre-Dataverse 4 times; back then harvested datasets ("studies") were similarly tied to their "harvesting dataverses". (A harvesting dataverse could only contain harvested studies, all harvested from the same place). That old query is still producing  the expected results in our prod., because we still have dedicated collections for each harvested source. Strictly speaking, the dvobject-dataset joint in it was useful,  although `o.dtype='Dataset'` would achieve the same result.  But I just remembered that this was no longer required. In Dataverse 4+ it is possible to harvest into a collection that has other content - local datasets and/or datasets harvested from other places. So, shorter answer,  yes, that query was wrong.